### PR TITLE
fix: compare siblings with real filepath instead of virtual in `unique-fragment-name` and `unique-operation-name`

### PR DESCRIPTION
**Pull request by B2o5T**
Related: https://github.com/dotansimha/graphql-eslint/issues/515

- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: docs/rules/no-hashtag-description.md [hunk 1]
```diff
@@ -6,7 +6,7 @@
 - Requires GraphQL Operations: `false` [ℹ️](../../README.md#extended-linting-rules-with-siblings-operations)
 
 Requires to use `"""` or `"` for adding a GraphQL description instead of `#`.
-This rule allows you to use hashtag for comments, as long as it's not attached to a AST definition.
+Allows to use hashtag for comments, as long as it's not attached to an AST definition.
```
Comment (B2o5T) [inline docs/rules/no-hashtag-description.md:9]: remove redundant `This rule allows you`
image

File: packages/plugin/src/processors/code-files.ts [hunk 2]
```diff
@@ -9,13 +10,12 @@ type Block = {
   offset: number;
 };
 
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
-export function createGraphqlProcessor() {
+export function createGraphqlProcessor(): Linter.Processor {
```
Comment (B2o5T) [inline packages/plugin/src/processors/code-files.ts:13]: improve types

File: packages/plugin/src/rules/unique-fragment-name.ts [hunk 3]
```diff
@@ -1,16 +1,64 @@
...
+import { GraphQLESLintRule, GraphQLESLintRuleContext } from '../types';
+import { GraphQLESTreeNode } from '../estree-parser';
+import { normalizePath, requireSiblingsOperations, getOnDiskFilepath } from '../utils';
+import { FragmentSource, OperationSource } from '../sibling-operations';
 
+const RULE_NAME = 'unique-fragment-name';
 const UNIQUE_FRAGMENT_NAME = 'UNIQUE_FRAGMENT_NAME';
 
-const rule: GraphQLESLintRule<[], false> = {
+export const checkNode = (
```
Comment (B2o5T) [inline packages/plugin/src/rules/unique-fragment-name.ts:11]: I removed code duplication in `unique-fragment-name` and `unique-operation-name` rules, moved logic in this `checkNode` function

File: scripts/generate-docs.ts [hunk 4]
```diff
@@ -118,7 +120,7 @@ async function main() {
       BR,
       'Each rule has emojis denoting:',
       BR,
-      `* ${EMOJI_GRAPHQL_ESLINT} \`@graphql-eslint\` rule`,
+      `* ${EMOJI_GRAPHQL_ESLINT} \`graphql-eslint\` rule`,
```
Comment (B2o5T) [inline scripts/generate-docs.ts:123]: `graphql-eslint` is more correct

**Review by dotansimha — APPROVED** [review]

**Pull request merged**